ui/packages/catalog-ui-search/src/main/webapp/react-component/menu/menu.tsx

```diff
@@ -249,5 +248,78 @@ export const MenuItem = (props: MenuItemProps) => {
   )
 }
 
+type MenutItemPropsDisabled = {
+  /** A value to represent the current Item */
+  value?: any
+  /**
+   * Children to display for menu item.
+   *
+   * @default value
+   */
+  children?: any
+  /** Optional styles for root element. */
+  style?: object
+  onClick?: any
+  selected?: any
+  active?: any
+  onHover?: any
+  disabled?: any
+  title?: any
+}
+
+const ItemRootDisabled = styled.option<{ active: boolean; disabled: boolean }>`
+  position: relative;
+  padding: 0px ${({ theme }) => theme.minimumSpacing};
+  padding-right: ${({ theme }) => theme.minimumButtonSize};
+  box-sizing: border-box;
+  height: ${({ theme }) => theme.minimumButtonSize};
+  line-height: ${({ theme }) => theme.minimumButtonSize};
+  cursor: pointer;
+  -webkit-touch-callout: none; /* iOS Safari */
+  -webkit-user-select: none; /* Safari */
+  -khtml-user-select: none; /* Konqueror HTML */
+  -moz-user-select: none; /* Firefox */
+  -ms-user-select: none; /* Internet Explorer/Edge */
+  user-select: none; /* Non-prefixed version, currently supported by Chrome and Opera */
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  ${({ theme, active, disabled }) =>
+    active && !disabled
+      ? `box-shadow: inset 0px 0px 0px 1px  ${theme.primaryColor};`
+      : ''}
+  ${({ active }) => (active ? 'font-weight: bold;' : '')}
+  ${({ active, disabled }) => (active && !disabled ? after : '')}
+  background: ${props =>
+    props.active && !props.disabled ? background(props) : 'inherit'};
+  color: ${props => (props.disabled ? 'lightgrey' : foreground)};
+`
+export const MenuItemDisabled = (props: MenutItemPropsDisabled) => {
+  const {
+    value,
+    children,
+    onClick,
+    active,
+    onHover,
+    style,
+    disabled,
+    title,
+  } = props
+  return (
+    <ItemRootDisabled
+      disabled={disabled}
+      active={active}
+      style={style}
+      onMouseEnter={() => onHover(value)}
+      onFocus={() => onHover(value)}
+      tabIndex={0}
+      onClick={() => onClick(value)}
+      title={title}
+    >
+      {children || value}
+    </ItemRootDisabled>
+  )
+}
+
```

Comment:
Yes I can but what I was going for here was to define a MenuItem that's disabled seperately so who ever would like to use it can do so without impacting other MenuItem's that's used in GSR.

Reply:
Disabled added to MenuItem  👍